### PR TITLE
Fix: The value of the AttributeCommand is taken from the first allowed node

### LINKED ISSUE #56: Weird behavior if selection starts in an element which disallows given attribute
Consider following scenario:
```html
<h1>Tit[le</h1>
<p>F]oo bar.</p>
```

Selection starts in heading and ends in paragraph. Assume that `bold` is disallowed in heading.

As of now, you are able to bold this selection, but you are unable to unbold it. This is because command's `value` is taken from first position but `isEnabled` from whole selection. So before you bold, command is enabled, so you bold the selection. But, the part in heading is not bold. So the `value` is still `false` (not bolded). Using the command again does not change anything because again and again bold is applied.

As much as we had good reasons for implementing those properties as we did, the behavior seems strange and buggy. We need to change it. We should either scan the whole range when checking `value` or change how `isEnabled` works.

Right now `isEnabled` is `true` if there is at least one place in selection where an attribute can be applied. Maybe we should change so that it is `false` whenever there is at least one place in selection where an attribute cannot be applied. Or make it `false` if selection starts in a place where the attribute cannot be applied.

### PR DESCRIPTION
**Pull request by oskarwrobel**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: The value of the `AttributeCommand` is taken from the first allowed node. Closes #56.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

File: src/attributecommand.js [hunk 1]
```diff
@@ -108,4 +108,39 @@ export default class AttributeCommand extends Command {
...
+		const selection = model.document.selection;
+
+		if ( selection.isCollapsed ) {
+			return selection.hasAttribute( this.attributeKey );
+		}
+
+		const ranges = Array.from( selection.getRanges() );
+
+		if ( selection.isBackward ) {
+			ranges.reverse();
```
Comment (pjasiun) [inline src/attributecommand.js]: Are you sure it is needed here?

File: src/attributecommand.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ export default class AttributeCommand extends Command {
 		 * Flag indicating whether the command is active. The command is active when the
 		 * {@link module:engine/model/selection~Selection#hasAttribute selection has the attribute} which means that:
 		 *
-		 * * If the selection is not empty &ndash; That it starts in a text (or another node) which has the attribute set.
+		 * * If the selection is not empty &ndash; That the first node in the selection that allows the attribute has the attribute set.
```
Comment (pjasiun) [inline src/attributecommand.js]: > the selection that allows the attribute has the attribute set

Something is wrong in this sentence.

File: src/attributecommand.js (hunk 1 above)
Comment (oskarwrobel) [reply src/attributecommand.js]: Hm...

Comment (oskarwrobel) [reply src/attributecommand.js]: Checking range direction is removed.

File: src/attributecommand.js [hunk 3]
```diff
@@ -108,4 +108,33 @@ export default class AttributeCommand extends Command {
 			}
 		} );
 	}
+
+	/**
+	 * Checks the attribute value of the first node in the selection that allows the attribute.
+	 * For the collapsed selection returns the selection attribute.
+	 *
+	 * **Note** Selection direction is taken into consideration.
```
Comment (pjasiun) [inline src/attributecommand.js]: No, it's not anymore ;)

Comment (oskarwrobel) [reply src/attributecommand.js]: Right :)

**Pull request merged**